MyString.java

```diff
@@ -64,8 +97,14 @@ public static String spacedString(String str) {
      * @return a randomly generated string, consisting of 'n' lowercase letters
      */
     public static String randomStringOfLetters(int n) {
-        //// Replace the following statement with your code
-        return null;
+        String randStr = "";
+        
+        for (int i = 0; i < n; i++) {
+            int randInt = (int) (Math.random() * 26); // random number from 0-25, covering all ASCII 97-122
+            randStr += (char) (97 + randInt);

```

Comment:
1. 26 is magic value .. use constant or array's length  
2. 97 is magic value .. Use ASCII

Reply:
Ok, got it for the first one, but why is 97 a magic value? I am adding 97 to the integer and casting it to char so that it produces a lowercase character between 'a' and 'z'

Reply:
"Magic numbers" or "Magic values" is a term that refers to unexplained strings or numbers which have meaning in your code.
If a random person will look at your code, without context on the assignment and tasks requested.
It will be strange. 
Hence you should avoid in few ways, 
1. by using a global variable for example (if this was in Scrabble.java) : `int randInt = (int) (Math.random() * SCRABBLE_LETTER_VALUES.length)`
2. by using more detailed data :` randStr += (char) ('a' + randInt);` or `int randInt = (int) (Math.random() * ('z' - 'a' + 1)));`
3. by introducing a constant variable :
```
public static final int NUM_LETTERS_IN_ENGLISH_ALPHABET = 26
int randInt = (int) (Math.random() * NUM_LETTERS_IN_ENGLISH_ALPHABET)
```